### PR TITLE
Camera: Option to disable autoconnect

### PR DESCRIPTION
**Pull request by sjkummer**
This introduces a new option to disable autoconnect when creating a camera object:

` * @param {boolean} [options.autoconnect=true] Set false if the camera should not connect automatically. The callback will not be executed.`

**Background / intent:** 
Instantiating a camera object automatically triggers multiple Onvif requests. This puts load on the camera. If the camera has to serve multiple clients, the Onvif API may get very slow, resulting in timeouts.

It should be possible, to perform a lean connectivity check, eg. calling `getSystemDateAndTime` without further overhad.

However, the default behaviour remains the same before and after this change. It just introduces an additional option parameter.

### CODE REVIEW

**Comment by sjkummer** [comment]
@agsh Ready for merge?

**Comment by chriswiggins** [comment]
Can you please add tests to this and remove the change to that other line?

**Comment by sjkummer** [comment]
@chriswiggins Ok, done

File: test/common.coffee [hunk 1]
```diff
@@ -23,6 +23,17 @@ describe 'Common functions', () ->
...
 
+  describe 'autoconnect disabled', () ->
+    it 'should not connect automatically', (done) ->
+      options = {
+        autoconnect: false
+        timeout: 0
+      }
+      testcam = new onvif.Cam options, assert.fail
+      setTimeout () ->
+        done()
```
Comment (chriswiggins) [inline test/common.coffee:34]: Can you check here that it didn't actually try and connect?

File: test/common.coffee [hunk 2]
```diff
@@ -23,6 +23,17 @@ describe 'Common functions', () ->
       assert.equal defaultCam.path, '/onvif/device_service'
       done()
 
+  describe 'autoconnect disabled', () ->
```
Comment (chriswiggins) [inline test/common.coffee:26]: Please add another test where the `autoconnect` option isn't passed in and it **does**  connect

Comment (sjkummer) [reply test/common.coffee:31]: ok, done

File: test/common.coffee (hunk 1 above)
Comment (sjkummer) [reply test/common.coffee:40]: in this case, assert.fail() will be called. I've improved the readability now.

**Comment by chriswiggins** [comment]
Thanks for your contribution @sjkummer ! @agsh will release this in the next cycle.

**Pull request merged**